test/integration/authorization_test.go

```diff
@@ -1562,23 +1561,43 @@ func TestClusterPolicyCache(t *testing.T) {
 	if err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
-	// turn off rate limiting so cache misses are more likely
-	clusterAdminClientConfig.Burst = 9999999
-	clusterAdminClientConfig.QPS = 9999999
-	clusterAdminAuthorizationClient := authorizationclient.NewForConfigOrDie(clusterAdminClientConfig)
 
-	for i := 0; i < 100; i++ { // usually takes less than 60 attempts for this to cache miss
-		clusterRole := &authorizationapi.ClusterRole{ObjectMeta: metav1.ObjectMeta{GenerateName: time.Now().String()}}
-		clusterRole, err = clusterAdminAuthorizationClient.ClusterRoles().Create(clusterRole)
-		if err != nil {
-			t.Fatalf("unexpected error creating cluster role %q: %v", clusterRole.Name, err)
-		}
-		clusterRoleBinding := &authorizationapi.ClusterRoleBinding{
-			ObjectMeta: metav1.ObjectMeta{Name: clusterRole.Name},
-			Subjects:   []kapi.ObjectReference{{Name: "user", Kind: authorizationapi.UserKind}},
-			RoleRef:    kapi.ObjectReference{Name: clusterRole.Name}}
-		if _, err := clusterAdminAuthorizationClient.ClusterRoleBindings().Create(clusterRoleBinding); err != nil {
-			t.Fatalf("cache error creating cluster role binding %d %q: %v", i, clusterRoleBinding.Name, err)
+	// this client has an API token so it is safe
+	userClient, _, err := testutil.GetClientForUser(clusterAdminClientConfig, "user")
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// this client has no API token so it is unsafe (like a browser)
+	anonymousConfig := rest.AnonymousClientConfig(clusterAdminClientConfig)
+	anonymousConfig.ContentConfig.GroupVersion = &schema.GroupVersion{}
+	anonymousConfig.ContentConfig.NegotiatedSerializer = kapi.Codecs
+	anonymousClient, err := rest.RESTClientFor(anonymousConfig)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	for _, tc := range []struct {
+		name   string
+		client rest.Interface
+
+		expectedError string
+	}{
+		{
+			name:          "safe to proxy",
+			client:        userClient.Core().RESTClient(),
+			expectedError: `User "user" cannot get pods/proxy in the namespace "ns": User "user" cannot get pods/proxy in project "ns"`,
+		},
+		{
+			name:          "unsafe to proxy",
+			client:        anonymousClient,
+			expectedError: `User "system:anonymous" cannot get pods/proxy in the namespace "ns": User "system:anonymous" cannot get pods/unsafeproxy in project "ns"`,
```

Comment:
hardcoding the full error will be fragile... check for `pods/unsafeproxy` in one, and absence of `pods/unsafeproxy` in the other?

Reply:
also check leading `proxy` verb?